config/initializers/diaspora_federation.rb

```diff
@@ -61,5 +35,26 @@
 
       person_entity.save!
     end
+
+    on :fetch_private_key_by_id do |diaspora_id|
+      Person.find_by(diaspora_handle: diaspora_id).owner.encryption_key
+    end
+
+    on :fetch_private_key_by_post_guid do |guid|
+      author_user = Post.find_by(guid: guid).author.owner
+      author_user.encryption_key unless author_user.nil?
+    end
+
+    on :fetch_public_key_by_id do |diaspora_id|
+      Person.find_by(diaspora_handle: diaspora_id).author.public_key
+    end
+
+    on :fetch_public_key_by_post_guid do |guid|
+      Post.find_by(guid: guid).author.public_key
+    end
+
+    on :post_author_is_local? do |guid|
+      Post.find_by(guid: guid).author.local?
```

Comment:
Is the post guid really all the information we have at this point in the gem? I think this hook is highly confusing, I'd prefer much much more to yield the diaspora id of the author. Actually same applies to the other two hooks.